src/exe/main.cc

```diff
@@ -440,40 +406,50 @@ static void start_profiler(DDProfContext *ctx) {
 
 /**************************** Program Entry Point *****************************/
 int main(int argc, char *argv[]) {
-  DDProfContext ctx;
+  ddprof::CommandLineWrapper cmd_line({});
+  {
+    // Use a dynamic allocation to allow clean up
+    // in other exit flows
+    DDProfContext *ctx = new DDProfContext();
```

Comment:
Why not make ctx a unique_ptr ?

Reply:
It was not clear how to model the ownership considering the different forks.
I would have to move the context within the start_profiler function. The start profiler would return the unique pointer to model the fact it can possibly give us back the ownership.
I think this is a good idea, I had not finished thinking about it.

Reply:
I will defer this to another PR.